### ISSUE TITLE
Support building `DistinguishedName` from RFC 4514 input

### ISSUE THREAD

**Issue by natevw** [apple/swift-certificates#200]
It would be helpful if `DistinguishedName` (and perhaps its subcomponents?) could support initialization from [RFC 4514](https://www.rfc-editor.org/rfc/rfc4514)–formatted strings.

`RelativeDistinguishedName.Attribute` and its `.Value` have a rough sort of reverse form of this in https://github.com/apple/swift-certificates/blob/a846fc71499526eeceae5e3b739235ff249d4433/Sources/X509/RDNAttribute.swift#L211-L233 and https://github.com/apple/swift-certificates/blob/a846fc71499526eeceae5e3b739235ff249d4433/Sources/X509/RDNAttribute.swift#L148C46-L150 and this is highlighted in the docs https://github.com/apple/swift-certificates/blob/a846fc71499526eeceae5e3b739235ff249d4433/Sources/X509/Docs.docc/Examining%20Certificates.md?plain=1#L65-L66 as following the standard, but there is currently no parsing/deserialization offered in the other ddirection

**Comment by Lukasa** [comment]
Yeah, this sounds like a reasonable feature. I'd be happy to see someone take a swing at implementing this.
